Index: go/src/cmd/go/build.go
===================================================================
--- go.orig/src/cmd/go/build.go
+++ go/src/cmd/go/build.go
@@ -490,6 +490,9 @@ See also: go build, go get, go clean.
 func libname(args []string) string {
 	var libname string
 	for _, arg := range args {
+		if libname == "" && arg != "./..." && strings.HasSuffix(arg, "/...") {
+			libname = strings.Replace(strings.TrimSuffix(arg, "/...") + "_all", "/", "-", -1)
+		}
 		arg = strings.TrimSuffix(arg, "/...")
 		arg = strings.Replace(arg, "/", "-", -1)
 		if libname == "" {
