Index: git/src/go/build/build.go
===================================================================
--- git.orig/src/go/build/build.go	2015-12-29 07:32:55.437430396 -0800
+++ git/src/go/build/build.go	2015-12-29 07:32:55.437430396 -0800
@@ -1472,7 +1472,7 @@
 }
 
 // ToolDir is the directory containing build tools.
-var ToolDir = filepath.Join(runtime.GOROOT(), "pkg/tool/"+runtime.GOOS+"_"+runtime.GOARCH)
+var ToolDir = envOr("GOTOOLDIR", filepath.Join(runtime.GOROOT(), "pkg/tool/"+runtime.GOOS+"_"+runtime.GOARCH))
 
 // IsLocalImport reports whether the import path is
 // a local import path, like ".", "..", "./foo", or "../foo".
Index: git/src/cmd/go/build.go
===================================================================
--- git.orig/src/cmd/go/build.go	2015-12-29 07:32:55.437430396 -0800
+++ git/src/cmd/go/build.go	2015-12-29 07:32:55.437430396 -0800
@@ -1403,7 +1403,7 @@
 		}
 
 		cgoExe := tool("cgo")
-		if a.cgo != nil && a.cgo.target != "" {
+		if a.cgo != nil && a.cgo.target != "" && os.Getenv("GOTOOLDIR") == "" {
 			cgoExe = a.cgo.target
 		}
 		outGo, outObj, err := b.cgo(a.p, cgoExe, obj, pcCFLAGS, pcLDFLAGS, cgofiles, gccfiles, cxxfiles, a.p.MFiles)
