Index: git/src/cmd/go/build.go
===================================================================
--- git.orig/src/cmd/go/build.go
+++ git/src/cmd/go/build.go
@@ -536,9 +536,11 @@ func libname(args []string, pkgs []*Pack
 		}
 	}
 	var haveNonMeta bool
-	for _, arg := range args {
+	for i, arg := range args {
 		if isMetaPackage(arg) {
 			appendName(arg)
+		} else if (i == 0 && arg != "./..." && strings.HasSuffix(arg, "/...")) {
+			appendName(strings.Replace(strings.TrimSuffix(arg, "/..."), "/", "-", -1) + "_all")
 		} else {
 			haveNonMeta = true
 		}
